test/com/amazon/ion/streaming/ReaderTest.java

```diff
@@ -178,6 +178,45 @@ public void testSymbolValueOnNonSymbol()
         }
     }
 
+    @Test
+    public void testReadingDecimalAsInteger()
+    {
+        // decimal value <space> int conversion
+        read(
+            "0. 0 " +
+            "-2147483648. -2147483648 " + // Min int
+            "2147483647. 2147483647 " // Max int
+        );
+        while (in.next() != null)
+        {
+            int actual = in.intValue();
+
+            in.next();
+            int expected = in.intValue();
+
+            assertEquals(expected, actual);
+        }
+    }
```

Comment:
I agree that it's an awkward pattern, and I like your suggestion. I've opened #380 for this.

Reply:
In case you missed it, this is completely implemented in that gist. I'll add a comment in #380, maybe just fire out another PR after this one is shipped then.

Reply:
I didn't miss it; I just don't want to further inflate this PR. A follow-on PR would be great.